### PR TITLE
W-13673062: Update to Java 17

### PR DESCRIPTION
**Pull request by tomasperezc**
- Moved from JDK8 to JDK17
- Updated to build with Maven Wrapper 3.9.3
- Updated Ubuntu image to 22.04
- Updated `jackson-databind` to 2.13.5 (See (PR#73)[https://github.com/mulesoft/api_project_maven_client/pull/72])
- Set `maven-compiler-plugin` 3.11.0 at root pom, and remove redundant configuration in `polyglot-exchange`
- Bump to release version 2.3.0

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -132,6 +130,18 @@
...
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>3.11.0</version>
+                    <configuration>
+                        <properties>
+                            <source>1.8</source>
+                            <target>1.8</target>
+                            <release>8</release>
+                        </properties>
```
Comment (mdaloia) [inline pom.xml:142]: The `source`, `target` and `release` go directly under `configuration`:
https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html

**Review by mdaloia — CHANGES_REQUESTED** [review]

**Comment by marinarv** [comment]
Hello guys! I would suggest to release this lib with a change in the major version. ACB is not using java 17 in the short time and if we need to add some change which also required a release we will need to branch from this point (now there is not a special requirement but it could happen during the time we are not updated to java 17). Make sense?
@mdaloia @tomasperezc

**Comment by mdaloia** [comment]
> Hello guys! I would suggest to release this lib with a change in the major version. ACB is not using java 17 in the short time and if we need to add some change which also required a release we will need to branch from this point (now there is not a special requirement but it could happen during the time we are not updated to java 17). Make sense? @mdaloia @tomasperezc

@marinarv we are not dropping support to Java 8 or 11... in fact we are compiling with source, target and release using language level 1.8.... that is the reason why we didn't released a major version.
You shouldn't face any issue using it with 8 or 11... let us know if you found some issues.

**Comment by marinarv** [comment]
> 

If that is the case we are ok! Thank you!

**Pull request merged**